### PR TITLE
Reducing the complexity of the examples

### PR DESCRIPTION
**Pull request by carlosypunto**
Reducing the complexity of the examples, centralizing dependences in a singleton.

### CODE REVIEW

**Comment by kzaher** [comment]
I'll try to develop this project using gitflow workflow http://nvie.com/posts/a-successful-git-branching-model/

I would appreciate if you could make this PR for develop branch. Master branch should contain latest stable tested version of the code only.

I have violated those principles when updating the documentation because there is significant value in making project more understandable with no risk attached, but need to be careful with any code change.

File: RxExample/RxExample/Examples/WikipediaImageSearch/ViewModels/SearchResultViewModel.swift [hunk 1]
```diff
@@ -16,12 +16,8 @@ class SearchResultViewModel {
     var title: Observable<String>
     var imageURLs: Observable<[NSURL]>
     
-    var $: SearchViewModel.Dependencies
```
Comment (kzaher) [inline RxExample/RxExample/Examples/WikipediaImageSearch/ViewModels/SearchResultViewModel.swift]: I see how removing a layer of boilerplate code could increase readability.

How do you feel about making $ a getter only variable to dependency object.
var $: Dependencies {
     get {
            return Dependencies.sharedDependencies
     }
}

I feel like adding Dependencies.sharedDependencies across the code is duplicating information and adds a lot of visual noise IMHO.

File: RxExample/RxExample/Examples/WikipediaImageSearch/ViewModels/SearchViewModel.swift [hunk 2]
```diff
@@ -11,47 +11,29 @@ import RxSwift
...
-        backgroundWorkScheduler: ImmediateScheduler,
-        mainScheduler: DispatchQueueScheduler,
-        wireframe: Wireframe
-    )
     
     // outputs
     let rows: Observable<[SearchResultViewModel]>
     
-    var $: Dependencies
-
```
Comment (kzaher) [inline RxExample/RxExample/Examples/WikipediaImageSearch/ViewModels/SearchViewModel.swift]: The same is valid here.

File: RxExample/RxExample/Examples/WikipediaImageSearch/Views/WikipediaSearchCell.swift [hunk 3]
```diff
@@ -27,8 +27,6 @@ public class WikipediaSearchCell: UITableViewCell {
     
     var viewModel: SearchResultViewModel! {
         didSet {
-            let $ = viewModel.$
```
Comment (kzaher) [inline RxExample/RxExample/Examples/WikipediaImageSearch/Views/WikipediaSearchCell.swift]: This could be turned into a getter also

File: RxExample/RxExample/Examples/WikipediaImageSearch/ViewModels/SearchResultViewModel.swift (hunk 1 above)
Comment (carlosypunto) [inline RxExample/RxExample/Examples/WikipediaImageSearch/ViewModels/SearchResultViewModel.swift]: I saw interesting the separation o the dependencies creatión

File: RxExample/RxExample/Examples/WikipediaImageSearch/ViewModels/SearchViewModel.swift (hunk 2 above)
Comment (carlosypunto) [inline RxExample/RxExample/Examples/WikipediaImageSearch/ViewModels/SearchViewModel.swift]: and also I see it

File: RxExample/RxExample/Examples/WikipediaImageSearch/Views/WikipediaSearchCell.swift (hunk 3 above)
Comment (carlosypunto) [inline RxExample/RxExample/Examples/WikipediaImageSearch/Views/WikipediaSearchCell.swift]: And also I see it. You want me to do another PR?

**Comment by carlosypunto** [comment]
I'm sorry

**Comment by carlosypunto** [comment]
I close it. In short I will send another proposal with what you comment in the correct branch. I apologize for the noise in the repository
